downloader/src/main/kotlin/VcsHost.kt

```diff
@@ -307,6 +305,11 @@ enum class VcsHost(
             return values().find { host -> host.isApplicable(vcsInfo) }
                 ?.toPermalinkInternal(vcsInfo.normalize(), startLine, endLine)
         }
+
+        /**
+         * Return the [VcsHost] for a [vcsUrl]
+         */
+        fun toVcsHost(vcsUrl: String): VcsHost? = values().find { host -> host.isApplicable(URI(vcsUrl)) }
```

Comment:
I was about to argue against turning `vcsUrl` to a `String` because now a possible `URISyntaxException` isn't caught when only using this function. In general, if a thing is a URI I'd vote for dealing with it as a URI as long as we can. Similarly, I'd treat a path as a `File` instead of a `String` as long as we can, just to name an example.

Reply:
I agree. Reverted my change and `toVcsHost` now again uses an `URI`.
@fviernau are you ok with that?